### ISSUE TITLE
Fails to build.

### ISSUE THREAD

**Issue by W-Floyd** [linuxdeepin/dde-daemon#51]
I believe this is more to do with upstream dependency changes, as it stopped building without a version upgrade. Running on Solus, everything up to date.

```
env GOPATH="/home/build/YPKG/root/deepin-daemon/build/dde-daemon.git/gopath:/home/build/YPKG/root/deepin-daemon/build/dde-daemon.git/:/usr/share/gocode" go build -o out/bin/default-terminal  pkg.deepin.io/dde/daemon/bin/default-terminal
gcc bin/default-file-manager/main.c -pthread -I/usr/include/gio-unix-2.0/ -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -lgio-2.0 -lgobject-2.0 -lglib-2.0 -o out/bin/default-file-manager
gcc bin/desktop-toggle/main.c -lX11 -o out/bin/desktop-toggle
env GOPATH="/home/build/YPKG/root/deepin-daemon/build/dde-daemon.git/gopath:/home/build/YPKG/root/deepin-daemon/build/dde-daemon.git/:/usr/share/gocode" go build -o out/bin/dde-session-initializer  pkg.deepin.io/dde/daemon/bin/dde-session-initializer
env GOPATH="/home/build/YPKG/root/deepin-daemon/build/dde-daemon.git/gopath:/home/build/YPKG/root/deepin-daemon/build/dde-daemon.git/:/usr/share/gocode" go build -o out/bin/dde-session-daemon  pkg.deepin.io/dde/daemon/bin/dde-session-daemon
env GOPATH="/home/build/YPKG/root/deepin-daemon/build/dde-daemon.git/gopath:/home/build/YPKG/root/deepin-daemon/build/dde-daemon.git/:/usr/share/gocode" go build -o out/bin/dde-system-daemon  pkg.deepin.io/dde/daemon/bin/dde-system-daemon
# pkg.deepin.io/dde/daemon/trayicon
gopath/src/pkg.deepin.io/dde/daemon/trayicon/daemon.go:59:18: undefined: ewmh.NewConn
gopath/src/pkg.deepin.io/dde/daemon/trayicon/daemon.go:64:19: undefined: icccm.NewConn
gopath/src/pkg.deepin.io/dde/daemon/trayicon/icon.go:70:24: XConn.GenerateID undefined (type *x.Conn has no field or method GenerateID)
gopath/src/pkg.deepin.io/dde/daemon/trayicon/init.go:42:13: undefined: ewmh.Conn
gopath/src/pkg.deepin.io/dde/daemon/trayicon/init.go:43:13: undefined: icccm.Conn
gopath/src/pkg.deepin.io/dde/daemon/trayicon/tools.go:36:17: vinfo.VisualId undefined (type x.VisualType has no field or method VisualId)
gopath/src/pkg.deepin.io/dde/daemon/trayicon/traymanager.go:195:15: unknown field 'ResponseType' in struct literal of type x.ClientMessageEvent
gopath/src/pkg.deepin.io/dde/daemon/trayicon/traymanager.go:202:2: undefined: x.ClientMessageEventWrite
gopath/src/pkg.deepin.io/dde/daemon/trayicon/traymanager.go:204:27: cannot use x.False (type int) as type bool in argument to x.SendEventChecked
gopath/src/pkg.deepin.io/dde/daemon/trayicon/traymanager.go:228:21: XConn.GenerateID undefined (type *x.Conn has no field or method GenerateID)
gopath/src/pkg.deepin.io/dde/daemon/trayicon/traymanager.go:228:21: too many errors
# pkg.deepin.io/dde/daemon/gesture
gopath/src/pkg.deepin.io/dde/daemon/gesture/utils.go:45:19: undefined: ewmh.NewConn
# pkg.deepin.io/dde/daemon/keybinding/shortcuts
gopath/src/pkg.deepin.io/dde/daemon/keybinding/shortcuts/shortcut_manager.go:59:21: undefined: ewmh.Conn
make: *** [Makefile:44: out/bin/dde-session-initializer] Error 2
make: *** Waiting for unfinished jobs....
# pkg.deepin.io/dde/daemon/system/gesture
core.c: In function ‘handle_gesture_events’:
core.c:214:10: error: ‘LIBINPUT_EVENT_GESTURE_TAP_BEGIN’ undeclared (first use in this function); did you mean ‘LIBINPUT_EVENT_GESTURE_SWIPE_BEGIN’?
     case LIBINPUT_EVENT_GESTURE_TAP_BEGIN:
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          LIBINPUT_EVENT_GESTURE_SWIPE_BEGIN
core.c:214:10: note: each undeclared identifier is reported only once for each function it appears in
core.c:216:10: error: ‘LIBINPUT_EVENT_GESTURE_TAP_END’ undeclared (first use in this function); did you mean ‘LIBINPUT_EVENT_GESTURE_TAP_BEGIN’?
     case LIBINPUT_EVENT_GESTURE_TAP_END:
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          LIBINPUT_EVENT_GESTURE_TAP_BEGIN
core.c: In function ‘handle_events’:
core.c:285:14: error: ‘LIBINPUT_EVENT_GESTURE_TAP_BEGIN’ undeclared (first use in this function); did you mean ‘LIBINPUT_EVENT_GESTURE_SWIPE_BEGIN’?
         case LIBINPUT_EVENT_GESTURE_TAP_BEGIN:
              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
              LIBINPUT_EVENT_GESTURE_SWIPE_BEGIN
core.c:286:14: error: ‘LIBINPUT_EVENT_GESTURE_TAP_UPDATE’ undeclared (first use in this function); did you mean ‘LIBINPUT_EVENT_GESTURE_SWIPE_UPDATE’?
         case LIBINPUT_EVENT_GESTURE_TAP_UPDATE:
              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
              LIBINPUT_EVENT_GESTURE_SWIPE_UPDATE
core.c:287:14: error: ‘LIBINPUT_EVENT_GESTURE_TAP_END’ undeclared (first use in this function); did you mean ‘LIBINPUT_EVENT_GESTURE_TAP_BEGIN’?
         case LIBINPUT_EVENT_GESTURE_TAP_END:{
              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
              LIBINPUT_EVENT_GESTURE_TAP_BEGIN
# pkg.deepin.io/dde/daemon/clipboard
gsd-clipboard-manager.c: In function ‘send_selection_notify’:
gsd-clipboard-manager.c:142:9: warning: ‘gdk_error_trap_push’ is deprecated: Use 'gdk_x11_display_error_trap_push' instead [-Wdeprecated-declarations]
         gdk_error_trap_push ();
         ^~~~~~~~~~~~~~~~~~~
In file included from /usr/include/gtk-3.0/gdk/gdk.h:50:0,
                 from gsd-clipboard-manager.c:35:
/usr/include/gtk-3.0/gdk/gdkmain.h:70:32: note: declared here
 void                           gdk_error_trap_push        (void);
                                ^~~~~~~~~~~~~~~~~~~
gsd-clipboard-manager.c:151:9: warning: ‘gdk_error_trap_pop_ignored’ is deprecated: Use 'gdk_x11_display_error_trap_pop_ignored' instead [-Wdeprecated-declarations]
         gdk_error_trap_pop_ignored ();
         ^~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/gtk-3.0/gdk/gdk.h:50:0,
                 from gsd-clipboard-manager.c:35:
/usr/include/gtk-3.0/gdk/gdkmain.h:75:32: note: declared here
 void                           gdk_error_trap_pop_ignored (void);
                                ^~~~~~~~~~~~~~~~~~~~~~~~~~
gsd-clipboard-manager.c: In function ‘finish_selection_request’:
gsd-clipboard-manager.c:171:9: warning: ‘gdk_error_trap_push’ is deprecated: Use 'gdk_x11_display_error_trap_push' instead [-Wdeprecated-declarations]
         gdk_error_trap_push ();
         ^~~~~~~~~~~~~~~~~~~
In file included from /usr/include/gtk-3.0/gdk/gdk.h:50:0,
                 from gsd-clipboard-manager.c:35:
/usr/include/gtk-3.0/gdk/gdkmain.h:70:32: note: declared here
 void                           gdk_error_trap_push        (void);
                                ^~~~~~~~~~~~~~~~~~~
gsd-clipboard-manager.c:178:9: warning: ‘gdk_error_trap_pop_ignored’ is deprecated: Use 'gdk_x11_display_error_trap_pop_ignored' instead [-Wdeprecated-declarations]
         gdk_error_trap_pop_ignored ();
         ^~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/gtk-3.0/gdk/gdk.h:50:0,
                 from gsd-clipboard-manager.c:35:
/usr/include/gtk-3.0/gdk/gdkmain.h:75:32: note: declared here
 void                           gdk_error_trap_pop_ignored (void);
                                ^~~~~~~~~~~~~~~~~~~~~~~~~~
gsd-clipboard-manager.c: In function ‘convert_clipboard_manager’:
gsd-clipboard-manager.c:424:25: warning: ‘gdk_error_trap_push’ is deprecated: Use 'gdk_x11_display_error_trap_push' instead [-Wdeprecated-declarations]
                         gdk_error_trap_push ();
                         ^~~~~~~~~~~~~~~~~~~
In file included from /usr/include/gtk-3.0/gdk/gdk.h:50:0,
                 from gsd-clipboard-manager.c:35:
/usr/include/gtk-3.0/gdk/gdkmain.h:70:32: note: declared here
 void                           gdk_error_trap_push        (void);
                                ^~~~~~~~~~~~~~~~~~~
gsd-clipboard-manager.c:436:25: warning: ‘gdk_error_trap_pop’ is deprecated: Use 'gdk_x11_display_error_trap_pop' instead [-Wdeprecated-declarations]
                         if (gdk_error_trap_pop () != Success)
                         ^~
In file included from /usr/include/gtk-3.0/gdk/gdk.h:50:0,
                 from gsd-clipboard-manager.c:35:
/usr/include/gtk-3.0/gdk/gdkmain.h:73:32: note: declared here
 G_GNUC_WARN_UNUSED_RESULT gint gdk_error_trap_pop         (void);
                                ^~~~~~~~~~~~~~~~~~
gsd-clipboard-manager.c:439:25: warning: ‘gdk_error_trap_push’ is deprecated: Use 'gdk_x11_display_error_trap_push' instead [-Wdeprecated-declarations]
                         gdk_error_trap_push ();
                         ^~~~~~~~~~~~~~~~~~~
In file included from /usr/include/gtk-3.0/gdk/gdk.h:50:0,
                 from gsd-clipboard-manager.c:35:
/usr/include/gtk-3.0/gdk/gdkmain.h:70:32: note: declared here
 void                           gdk_error_trap_push        (void);
                                ^~~~~~~~~~~~~~~~~~~
gsd-clipboard-manager.c:449:33: warning: ‘gdk_error_trap_pop’ is deprecated: Use 'gdk_x11_display_error_trap_pop' instead [-Wdeprecated-declarations]
                                 if (gdk_error_trap_pop () != Success) {
                                 ^~
In file included from /usr/include/gtk-3.0/gdk/gdk.h:50:0,
                 from gsd-clipboard-manager.c:35:
/usr/include/gtk-3.0/gdk/gdkmain.h:73:32: note: declared here
 G_GNUC_WARN_UNUSED_RESULT gint gdk_error_trap_pop         (void);
                                ^~~~~~~~~~~~~~~~~~
gsd-clipboard-manager.c: In function ‘convert_clipboard_target’:
gsd-clipboard-manager.c:558:25: warning: ‘gdk_error_trap_push’ is deprecated: Use 'gdk_x11_display_error_trap_push' instead [-Wdeprecated-declarations]
                         gdk_error_trap_push ();
                         ^~~~~~~~~~~~~~~~~~~
In file included from /usr/include/gtk-3.0/gdk/gdk.h:50:0,
                 from gsd-clipboard-manager.c:35:
/usr/include/gtk-3.0/gdk/gdkmain.h:70:32: note: declared here
 void                           gdk_error_trap_push        (void);
                                ^~~~~~~~~~~~~~~~~~~
gsd-clipboard-manager.c:578:25: warning: ‘gdk_error_trap_pop_ignored’ is deprecated: Use 'gdk_x11_display_error_trap_pop_ignored' instead [-Wdeprecated-declarations]
                         gdk_error_trap_pop_ignored ();
                         ^~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/gtk-3.0/gdk/gdk.h:50:0,
                 from gsd-clipboard-manager.c:35:
/usr/include/gtk-3.0/gdk/gdkmain.h:75:32: note: declared here
 void                           gdk_error_trap_pop_ignored (void);
                                ^~~~~~~~~~~~~~~~~~~~~~~~~~
make: *** [Makefile:44: out/bin/dde-session-daemon] Error 2
make: *** [Makefile:44: out/bin/dde-system-daemon] Error 2
```

**Comment by jouyouyun** [comment]
Compile with the dde-daemon (patch)[https://github.com/jouyouyun/tap-gesture-patches/raw/master/patches/dde-daemon_3.2.3.patch] or the libinput (patch)[https://github.com/jouyouyun/tap-gesture-patches/raw/master/patches/tap_patch_1.10.7.patch].

**Comment by W-Floyd** [comment]
Had tried that before, didn't fix it, but it's building with 3.2.21 now.
